### PR TITLE
Sync fixes from snapcraft-desktop-helpers (LP-1661626 & broken XDG link)

### PR DESCRIPTION
**Pull request by MarcusTomlinson**
- [x] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] Have you successfully run `./runtests.sh static`?
- [x] Have you successfully run `./runtests.sh tests/unit`?

-----

Bug report: https://bugs.launchpad.net/snapcraft/+bug/1661626
XDG link error: https://paste.ubuntu.com/p/5HHF5qwS2f/

Adapted from: https://github.com/ubuntu/snapcraft-desktop-helpers/pull/201

### CODE REVIEW

File: extensions/desktop/kde-neon/launcher-specific [hunk 1]
```diff
@@ -45,7 +45,7 @@ export VLC_PLUGIN_PATH="$SNAP_DESKTOP_RUNTIME/usr/lib/x86_64-linux-gnu/vlc/plugi
 export QTCHOOSER_NO_GLOBAL_DIR=1
 export QT_SELECT=5
 # qtchooser hardcodes reference paths, we'll need to rewrite them properly
-[ -d "$XDG_CONFIG_HOME/qtchooser" ] || mkdir -p "$XDG_CONFIG_HOME/qtchooser"
+[ -d "$XDG_CONFIG_HOME/qtchooser" ] || ensure_dir_exists "$XDG_CONFIG_HOME/qtchooser"
```
Comment (cjp256) [inline extensions/desktop/kde-neon/launcher-specific]: I understand that this is just a sync, but I do believe this is redundant?
I think `ensure_dir_exists "$XDG_CONFIG_HOME/qtchooser"` should be sufficient?

Thanks for the updates! :smiley:

Comment (MarcusTomlinson) [reply extensions/desktop/kde-neon/launcher-specific]: Well spotted! Fixed.

**Comment by sergiusens** [comment]
Mind splitting out "ensure_dir" into its own PR?

**Comment by MarcusTomlinson** [comment]
> Mind splitting out "ensure_dir" into its own PR?

Sure, here you go: https://github.com/snapcore/snapcraft/pull/2886

**Comment by sergiusens** [comment]
Imported from ubuntu/snapcraft-desktop-helpers#201

**Pull request merged**